### ISSUE TITLE
Girl specific opportunities

### ISSUE THREAD

**Issue by Pratishtha33** [Girl-Code-It/Opportunity-Calendar-Frontend#179]
### **Feature request**:

there are many opportunities that are only for girls, to empower and motivate them.  it will be easier for them to browse specific opportunities.

### **Is your feature request related to a problem? Please describe. (optional)**:

YES, as I am a fresher girl from tier-3 college so I browse opportunities specifically for girls and I found it difficult because it takes too much time to search on the whole internet, there is not one website that contains all.

### **Describe the solution you'd like**:

so, I want to create one page, especially for girls' specific opportunities.

### **Describe how this feature will be useful to our readers**:
there are many programs for only girls like the Women Tech Makers program, sheHacks(hackathon), and many more.

### **Additional context:**

<hr/>

**Note:**

- If you want to work on an issue, you should check if it has already been assigned to anyone.

**Comment by Manvityagi** [comment]
I like the feature, 
but can we just add a filter on top of the View Opportunities Page, like - `View opportunities for females only` and `Exclude female-only opportunities`? 

We can also have a dedicated page for female-only opportunities though but it will be 1 single page unlike 6 types of opportunities in the current model of the website. 
On that single page, we can have filters of Scholarship, Jobs, etc. 

I have a little more inclination towards the 2nd implementation because that way the general users won't have to worry about using the exclude-female-specific filter and it will also be a nice addition to the website. But I would love to hear other's thoughts too on this. 

CC: @abdus @vaishali614 @DeepanshiD

**Comment by abdus** [comment]
@Manvityagi I would go with the first option. `Female Only` is not a category (given that the categories are divided based on the type of the opportunity ... internship, full-time etc). it definitely is a filter tho. having a toggle button somewhere at the top would be enough. Additionally, we can add labels to female-only opportunities.

**Comment by Manvityagi** [comment]
> @Manvityagi I would go with the first option. `Female Only` is not a category (given that the categories are divided based on the type of the opportunity ... internship, full-time etc). it definitely is a filter tho. having a toggle button somewhere at the top would be enough. Additionally, we can add labels to female-only opportunities.

Fair Enough

**Comment by DeepanshiD** [comment]
Yes! Having filter is most optimum solution to the problem and it serves feature purpose too
CC @Manvityagi @abdus

**Comment by Pratishtha33** [comment]
@Manvityagi @abdus  after clicking on the toggle button(filter) Will it is on the same page or a different page. If different then can I work on this??

**Comment by abdus** [comment]
@Pratishtha33 keeping it on the same page should just work. you can do something like `opportunity.forFemale ? show(opportunity) : hide(opportunity)`

cc: @DeepanshiD @Manvityagi

**Comment by DeepanshiD** [comment]
@abdus I guess we can do it better with a small checkbox below search bar, writing "Female Opportunities" and it will serve purpose and come in first sight too. 
Whatsay? @Manvityagi

**Comment by abdus** [comment]
@DeepanshiD yeah, that is what I was suggesting.

**Comment by vaishali614** [comment]
@Pratishtha33 are you still interested in working on this issue?

**Comment by Pratishtha33** [comment]
@vaishali614 yes Ma'am, I am pretty interested in adding the  features to the website and want to work. I have the Google Generation coding test day after tomorrow. After that i will definately work with you 😇.

**Comment by vaishali614** [comment]
Great. I'm assigning the issue to you then.

**Comment by vaishali614** [comment]
@Pratishtha33 What's your progress on this?

**Comment by Pratishtha33** [comment]
@vaishali614 I'm sorry for being late, I will add my code by tomorrow. what should I name the file??

**Comment by Manvityagi** [comment]
> @vaishali614 I'm sorry for being late, I will add my code by tomorrow. what should I name the file??

You don't need a new file, 
Just a checkbox type thing on top pf opportunities which says "Underrepresented Groups", 
I think more text than this won't look good, 
Instead you can have an "i" in a circle(representing more information), clicking which shows "See Opportunities only for underrepresented groups like transgenders, women, LGBTQ, black people etc".

CC: @DeepanshiD @vaishali614 @abdus 
What do you suggest?

**Comment by DeepanshiD** [comment]
I think we can segregate women and underrepresented groups, and list all underrepresented groups in tooltip ('i' in circle which on hover states all groups). 
<> women <> underrepresented groups (i) 
Something like above layout
CC : @vaishali614 @Manvityagi

**Comment by Manvityagi** [comment]
@Pratishtha33 
![image](https://user-images.githubusercontent.com/32908093/115379809-837a0600-a1ef-11eb-922f-4d14e06ae671.png)

A toggle button like this with the label 'Only for Women'

Refer to the image attached for an idea.

**Comment by Pratishtha33** [comment]
@Manvityagi thank you for giving an accurate idea, I was confused about what should I add.

**Comment by vaishali614** [comment]
@Pratishtha33 any updates?

**Comment by vaishali614** [comment]
@Pratishtha33 Due to your inactiveness, I'm unassigning you from this issue.

**Comment by AyushPradhan9** [comment]
Hey, I am interested to commit to the issue but I have some problems understanding the requirements, do we only have to add toggle button "Only for Girls" or make it work as well.

**Comment by vaishali614** [comment]
@AyushPradhan9 The toggle button should be working as well. It should list all the opportunities specific to females within the selected opportunity.

Also, since you've already been assigned an issue, this issue can't be assigned to you. As soon as you're able to complete that one, you can comment again and start working on this one.

**Comment by vaishali614** [comment]
@AyushPradhan9 Your PR has been merged. You're free to work on this issue now. Please let me know if you're still interested in working on this issue, I'll assign it to you.

**Comment by AyushPradhan9** [comment]
Apologizes @vaishali614  mam, I can't take such a big issue as of now because of some other commitments. I will surely inform you if I got some time. Thank you.

**Comment by raks924** [comment]
Hey, can i try working on this issue?

**Comment by vaishali614** [comment]
@raks924 Any updates on PR?

**Comment by raks924** [comment]
ummm @vaishali614 i think now if i work this wont be counted in pr scoreboard ...ryte?

**Comment by Manvityagi** [comment]
> ummm @vaishali614 i think now if i work this wont be counted in pr scoreboard ...ryte?

Yeah, GSSOC is over. 
But open-source is not only about score boards and programs :)

**Comment by raks924** [comment]
ya that is true :) .. but rn cant work on an urgent basis on this ....have some prior clg engagements
